### PR TITLE
[TOAZ-219]Create Landing Zone API for listing resources by purpose

### PR DESCRIPTION
**Pull request by zykonda**

### CODE REVIEW

File: service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java [hunk 1]
```diff
@@ -131,10 +156,25 @@ public ApiAzureLandingZoneResourcesList listAzureLandingZoneResources(
...
-        .filter(r -> r.getPurpose().equals(purpose.toString()))
+    return listAzureLandingZoneResources(bearerToken, landingZoneId, purpose)
+        .getResources()
+        .stream()
         .flatMap(r -> r.getDeployedResources().stream())
-        .collect(Collectors.toList());
+        .toList();
+  }
+
+  LandingZonePurpose getLandingZonePurpose(String purpose) {
```
Comment (rtitle) [inline service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java:166]: Nit - make private?

**Comment by rtitle** [comment]
Few nitpicks, overall looks good. I can 👍 when out of draft state.

**Review by giventocode — APPROVED** [review]
LGTM

File: service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java [hunk 2]
```diff
@@ -137,13 +136,11 @@ private ApiDeleteAzureLandingZoneResult toApiDeleteAzureLandingZoneResult(
 
   public ApiAzureLandingZoneResourcesList listAzureLandingZoneResources(
       BearerToken bearerToken, UUID landingZoneId) {
-    features.azureEnabledCheck();
-    LandingZoneResourcesByPurpose groupedResources =
-        landingZoneService.listResourcesWithPurposes(bearerToken, landingZoneId);
-
     var result = new ApiAzureLandingZoneResourcesList().id(landingZoneId);
+    features.azureEnabledCheck();
```
Comment (sergiygetlin) [inline service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java:140]: This probably should be the first line?

**Review by sergiygetlin — APPROVED** [review]

**Pull request merged**